model/lifecycle.go

```diff
@@ -1314,6 +1296,58 @@ func createOneTask(id string, buildVarTask BuildVariantTaskUnit, project *Projec
 	return t, nil
 }
 
+func generateDistros(id string, buildVarTask BuildVariantTaskUnit, buildVariant *BuildVariant, project *Project, v *Version) (string, []string) {
+	var (
+		distroID      string
+		distroAliases []string
+	)
+
+	if len(buildVarTask.RunOn) > 0 {
+		distroID = buildVarTask.RunOn[0]
+
+		if len(buildVarTask.RunOn) > 1 {
+			distroAliases = append(distroAliases, buildVarTask.RunOn[1:]...)
+		}
+
+	} else if len(buildVariant.RunOn) > 0 {
+		distroID = buildVariant.RunOn[0]
+
+		if len(buildVariant.RunOn) > 1 {
+			distroAliases = append(distroAliases, buildVariant.RunOn[1:]...)
+		}
+	} else {
+		grip.Warning(message.Fields{
```

Comment:
This seems like a weird case that should also be an error (I know the code is originally written like this, but it probably shouldn't do this unless there's a special reason). Since a task like this won't be able to run anyways, can this just return an error rather than log a warning and continue?